subsys/bluetooth/audio/broadcast_source.c

```diff
@@ -151,7 +151,7 @@ static void broadcast_source_iso_disconnected(struct bt_iso_chan *chan, uint8_t
 
 	BT_DBG("stream %p ep %p reason 0x%02x", chan, ep, reason);
 
-	broadcast_source_set_ep_state(ep, BT_AUDIO_EP_STATE_IDLE);
+	broadcast_source_set_ep_state(ep, BT_AUDIO_EP_STATE_QOS_CONFIGURED);
```

Comment:
> I think the correct state would be codec_configured or idle in this case.

I've had the same thought. In the non-idle state, the broadcast source will however advertise the BASE. The BASE does mostly contain codec information, so the `Codec configured` state would fit that part. It does, however, also advertise the presentation delay which is a QoS value, which is why we set it to the `BT_AUDIO_EP_STATE_QOS_CONFIGURED` state, as the broadcast source must have configured the QoS at this point. 

> But for broadcast it is indeed different and I'm not sure where to look this up - it might actually be implementation specific how to handle it, but is it wise that zephyr uses the same state defines for unicast and broadcast?

The BAP spec actually defines the following state machine for the broadcast source:

![image](https://user-images.githubusercontent.com/2520413/159666605-285d1220-31fd-4d1b-bccc-19beb6bd247e.png)

No such state machine is defined for the broadcast sink however, but we are using the same states for the broadcast sink in Zephyr

Reply:
Ok BAP - I see.